bookmaker_titlepage.rb

```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
-  Mcmlln::Tools.makeDir(final_dir)
-  Mcmlln::Tools.makeDir(final_dir_images)
-end
+makeFolder(final_dir, 'create_final_dir')
+makeFolder(final_dir_images, 'create_final_dir_images')
 
 # create the logging dir if it doesn't exist yet
-unless Dir.exist?(logdir)
-  Mcmlln::Tools.makeDir(logdir)
-end
+makeFolder(logdir, 'create_logdir')
 
 # find any archived titlepage images
-
-oldtitlepage = ""
-
-oldepubtitlepage = File.join(coverdir, "epubtitlepage.jpg")
-oldpodtitlepage = File.join(coverdir, "titlepage.jpg")
-
-if File.file?(oldepubtitlepage)
-  puts "Found an archived EPUB titlepage"
-  oldtitlepage = oldepubtitlepage
-elsif File.file?(oldpodtitlepage)
-  puts "Found an archived POD titlepage"
-  oldtitlepage = oldpodtitlepage
-end
+oldtitlepage = checkForOldTitlepages(coverdir, 'check_for_old_titlepages')
 
 # find any new user-submitted titlepage images
-allimg = File.join(submitted_images, "*")
-etparr = Dir[allimg].select { |f| f.include?('epubtitlepage.')}
-ptparr = Dir[allimg].select { |f| f.include?('titlepage.')}
-
-newtitlepage = ""
-
-if etparr.any?
-  puts "Found a new EPUB titlepage"
-  newtitlepage = etparr.find { |e| /[\/|\\]epubtitlepage\./ =~ e }
-elsif ptparr.any?
-  puts "Found a new POD titlepage"
-  newtitlepage = ptparr.find { |e| /[\/|\\]titlepage\./ =~ e }
-end
+newtitlepage = checkForSubmittedTitlepages(submitted_images, 'check_for_submitted_titlepages')
 
 # if an epub-specific titlepage file has been submitted, use that;
 # otherwise use the new POD coverpage if it exists;
 # and if neither exists, we'll create the epubtitlepage, and set it to the final archival path.
 # (POD titlepage images should only be submitted manually by the user, never created programatically.)
 
-final_cover = oldtitlepage
-
-if File.file?(newtitlepage)
-  final_cover = newtitlepage
-end
-
-# set the default switch to generate the titlepage
-gen = false
+final_cover = setFinalCover(oldtitlepage, newtitlepage, 'set_final_cover_var')
 
 # Determine whether or not to generate a titlepage.
-# First check: if an epub titlepage has previously been generated, and no new image has been submitted by the user
-if File.file?(titlepagelog) and !File.file?(newtitlepage)
-  gen = true
-  Mcmlln::Tools.deleteFile(titlepagelog)
-  Mcmlln::Tools.deleteFile(oldtitlepage)
-# Then check: if a titlepage has previously been generated, but there IS a new image submitted by the user
-elsif File.file?(titlepagelog) and File.file?(newtitlepage)
-  gen = false
-  Mcmlln::Tools.deleteFile(titlepagelog)
-  Mcmlln::Tools.deleteFile(oldtitlepage)
-# Finally: if no titlepage has ever been generated, and no new image has been submitted,
-# and there is no existing image archived from a previous run
-elsif !File.file?(titlepagelog) and !File.file?(final_cover)
-  gen = true
-end
-
+gen = decideTitlepageGeneration(titlepagelog, newtitlepage, oldtitlepage, final_cover, 'determine_titlepage_generation')
+@log_hash['gen(erate_titlepage)_value'] = gen
```

Comment:
I was explaining what 'gen' meant to aid in readability of jsonlogfile, but trying to still show it was the value of 'gen'.  How about "gen_value_(generate_titlepage_bool)"?